test/unit/iteration_test.rb

```diff
@@ -414,6 +414,33 @@ def test_total_time_is_updated_for_failed_jobs
     end
   end
 
+  def test_each_iteration_instrumentation
+    notifications = []
+    ActiveSupport::Notifications.monotonic_subscribe("each_iteration.iteration") do |_, _, _, _, tags|
```

Comment:
Can we use [`ActiveSupport::Notifications.subscribed`](https://api.rubyonrails.org/classes/ActiveSupport/Notifications.html#module-ActiveSupport::Notifications-label-Temporary+Subscriptions) here, so the subscription is removed after the test finishes?